File: flutter_zoom_videosdk-2.4.0/android/src/main/java/com/flutterzoom/videosdk/FlutterZoomVideoSdkAudioSettingHelper.java
package com.flutterzoom.videosdk;

import android.app.Activity;

import androidx.annotation.NonNull;

import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkErrors;

import java.util.Map;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import us.zoom.sdk.ZoomVideoSDK;
import us.zoom.sdk.ZoomVideoSDKAudioSettingHelper;

public class FlutterZoomVideoSdkAudioSettingHelper {

    private Activity activity;

    FlutterZoomVideoSdkAudioSettingHelper(Activity activity) {
        this.activity = activity;
    }

    private ZoomVideoSDKAudioSettingHelper getAudioSettingHelper() {
        ZoomVideoSDKAudioSettingHelper audioSettingHelper = null;

        try {
            audioSettingHelper = ZoomVideoSDK.getInstance().getAudioSettingHelper();
            if (audioSettingHelper == null) {
                throw new Exception("No Audio Setting Helper Found");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return audioSettingHelper;
    }

    public void isMicOriginalInputEnable(@NonNull MethodChannel.Result result) {
        result.success(getAudioSettingHelper().isMicOriginalInputEnable());
    }

    public void enableMicOriginalInput(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> args = call.arguments();
        boolean enable = (Boolean) args.get("enable");
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getAudioSettingHelper().enableMicOriginalInput(enable)));
            }
        });
    }

    public void enableAutoAdjustMicVolume(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> args = call.arguments();
        boolean enable = (Boolean) args.get("enable");
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getAudioSettingHelper().enableAutoAdjustMicVolume(enable)));
            }
        });
    }

    public void isAutoAdjustMicVolumeEnabled(@NonNull MethodChannel.Result result) {
        result.success(getAudioSettingHelper().isAutoAdjustMicVolumeEnabled());
    }

}

File: flutter_zoom_videosdk-2.4.0/ios/Classes/FlutterZoomVideoSdkAudioSettingHelper.m
#import "FlutterZoomVideoSdkAudioSettingHelper.h"
#import "JSONConvert.h"

@implementation FlutterZoomVideoSdkAudioSettingHelper

- (ZoomVideoSDKAudioSettingHelper *)getAudioSettingHelper {
    ZoomVideoSDKAudioSettingHelper* audioSettingHelper = nil;
    @try {
        audioSettingHelper = [[ZoomVideoSDK shareInstance] getAudioSettingHelper];
        if (audioSettingHelper == nil) {
            NSException *e = [NSException exceptionWithName:@"NoAudioSettingHelperFound" reason:@"No Audio Setting Helper Found" userInfo:nil];
            @throw e;
        }
    } @catch (NSException *e) {
        NSLog(@"%@ - %@", e.name, e.reason);
    }
    return audioSettingHelper;
}

-(void) isMicOriginalInputEnable: (FlutterResult) result {
    if ([[self getAudioSettingHelper] isMicOriginalInputEnable]) {
        result(@YES);
    } else {
        result(@NO);
    }
}

-(void) enableMicOriginalInput: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getAudioSettingHelper] enableMicOriginalInput:[call.arguments[@"enable"] boolValue]])]);
    });

}

-(void) isAutoAdjustMicVolumeEnabled: (FlutterResult) result {
    if ([[self getAudioSettingHelper] isAutoAdjustMicVolumeEnabled]) {
        result(@YES);
    } else {
        result(@NO);
    }
}

-(void) enableAutoAdjustMicVolume: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getAudioSettingHelper] enableAutoAdjustMicVolume:[call.arguments[@"enable"] boolValue]])]);
    });

}

@end
